cmd/supernode/app/root.go

```diff
@@ -130,7 +130,13 @@ func runSuperNode() error {
 		return err
 	}
 
-	if err = d.Run(); err != nil {
+	// register supernode
+	if err := d.RegisterSuperNode(); err != nil {
+		logrus.Errorf("failed to register super node: %v", err)
+		return err
+	}
+
+	if err := d.Run(); err != nil {
 		logrus.Errorf("failed to run daemon: %v", err)
```

Comment:
There is a similar error log in function `d.Run()`, maybe it's better to remove it from `d.Run()`.

Reply:
Updated.